### PR TITLE
Add an introducer property to DeclGroupSyntax

### LINKED ISSUE #2535: DeclGroupSyntax should have introducerKeyword
### Description

It should be possible to `.as(DeclGroupSyntax.self)?.introducerKeyword` when writing macros in order to easily write diagnostics like "cannot be applied to **struct**"  etc.

Currently one has to do manual if/else of all the kinds we support

### Steps to Reproduce

_No response_

### PR DESCRIPTION
**Pull request by bnbarham**
It can be useful to grab the introducer of a particular declaration, add a property for this purpose to `DeclGroupSyntax`.

Fixes #2535.
Resolves rdar://124247480.

### CODE REVIEW

**Comment by bnbarham** [comment]
@swift-ci please test

File: Sources/SwiftSyntax/generated/SyntaxTraits.swift [hunk 1]
```diff
@@ -72,6 +72,22 @@ public protocol DeclGroupSyntax: SyntaxProtocol, DeclSyntaxProtocol {
     set
   }
```
Comment (bnbarham) [inline Sources/SwiftSyntax/generated/SyntaxTraits.swift:74]: Also open to renaming `DeclGroupSyntax` if anyone has a better name for it.

**Review by ktoso — APPROVED** [review]
Nice, thank you!

File: Sources/SwiftSyntax/CustomTraits.swift [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  var introducer: TokenSyntax {
+    get {
+      return extensionKeyword
+    }
+    set {
+      extensionKeyword = newValue
+    }
+  }
+}
+extension ProtocolDeclSyntax {
```
Comment (ahoppen) [inline Sources/SwiftSyntax/CustomTraits.swift:56]: Very nitpicky: Above you have a newline between the extensions, here you don’t.

File: Sources/SwiftSyntax/generated/SyntaxTraits.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/SwiftSyntax/generated/SyntaxTraits.swift:74]: I’m not a huge fan of my suggestion but what this conceptually is, is a `DeclWithMembersSyntax`, I think. At least that describes the trait IMO.

File: Sources/SwiftSyntax/CustomTraits.swift (hunk 2 above)
Comment (bnbarham) [reply Sources/SwiftSyntax/CustomTraits.swift:56]: Ah thanks, copy paste strikes again :)

File: Sources/SwiftSyntax/generated/SyntaxTraits.swift (hunk 1 above)
Comment (bnbarham) [reply Sources/SwiftSyntax/generated/SyntaxTraits.swift:74]: That matches the other "With" at least 😅

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by ahoppen — APPROVED** [review]

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please test

**Pull request merged**